### PR TITLE
Fix problems when some ArcGIS layers is loading

### PR DESCRIPTION
**Pull request by elesdoar**

### CODE REVIEW

File: src/services/leafletLayerHelpers.js [hunk 1]
```diff
@@ -472,8 +494,35 @@ angular.module("leaflet-directive")
         }
     }
 
+    function safeRemoveLayer(map, layer, layerOptions) {
```
Comment (nmccready) [inline src/services/leafletLayerHelpers.js]: :+1:

File: src/services/leafletLayerHelpers.js [hunk 2]
```diff
@@ -213,7 +223,19 @@ angular.module("leaflet-directive")
...
 
-                return L.esri.featureLayer(params.options);
+                var layer = L.esri.featureLayer(params.options);
+                var load = function() {
+                    if(isDefined(params.options.loadedDefer)) {
+                        params.options.loadedDefer.resolve();
+                    }
+                };
+                layer.on('loading', function() {
+                    params.options.loadedDefer = $q.defer();
```
Comment (nmccready) [inline src/services/leafletLayerHelpers.js]: Is this not dangerous to allow anyone to have access to loadedDefer to resolve it when they want to?

Comment (elesdoar) [inline src/services/leafletLayerHelpers.js]: Maybe, but, I not found another way to do it, but I need refresh the loadedDefer when loading event is fired.

Any idea?

Comment (nmccready) [inline src/services/leafletLayerHelpers.js]: eh.. oh well if someone wants to resolve it early then so be it; then it allows them to cancel it as well. I am not going to loose sleep over this.

**Comment by nmccready** [comment]
LGTM

**Comment by elesdoar** [comment]
:smile:

**Pull request merged**